### DISCUSSION TITLE
How to bundle using vite-node

### DISCUSSION THREAD

**Discussion by azarenkovgd** [vitest-dev/vitest#4420] [Q&A] [answered]
I see examples of dev setup here (https://github.com/vitest-dev/vitest/tree/main/packages/vite-node#readme). But there are no examples of bundling with vite-node. 
Also, there are no examples of using vite-node without HMR. I kind of got it working by tweaking default setup, but problem with bundling remains relevant.
I use vite on frontend and rollup on backend, vitest for testing. It would be great to use vite for backend also. 
Maybe it is incorrect requirement and vite-node is not suitable for this purpose. I know that vite is based on rollup, but it has some nice defaults.

So the final questions are:
1. How to bundle backend code using vite-node?
2. If it is not possible now, is it on the roadmap?

**Comment by sheremet-va** [comment, marked as answer]
vite-node is not a bundler, it's a runtime (you can think about it as just running `node index.js`), you can bundle with any available bundler - since vite-node uses Vite, you can add a build config there - https://vitejs.dev/guide/build.html

**Reply by raythurnvoid** [reply]
You should consider this since up to today there's no proper way to get started with TS. All other alternatives are somehow worse:

ts-node: slow, memory intensive, configuration is needed, if you use swc you give up on type checking.
tsx: easy to get started but you have to give up on type checking.
tsc: a lot of manual work to set up with nodemon or similar.

Since vite-node uses vite, you can easily import stuff like images, text files, binary files and it doesn't require any configuration to start.
The only issue I see is that running vite-node starts 2 node processes that summed take 6 times more memory than Node. So a "--production" mode that transpiles everything at start and then uses node would be great. It would be the closest thing to Deno that Node can have today.
